### PR TITLE
[AR-4416] Plug n Play Auth

### PR DESCRIPTION
**Pull request by makylfang**
# PR

## Describe your changes

- Added plug n play auth, can be called via `.connect()` function
- Added arcana provider to `window.ethereum.providers`
- Updated dependency for `ts-jest`
- Updated version
- Added `EthereumProvider` interface

## Screenshots:
Screen Shot 2022-11-03 at 10 28 47 AM
Screen Shot 2022-11-03 at 10 29 27 AM

## Issue ticket number and link

- [AR-4416](https://team-1624093970686.atlassian.net/browse/AR-4416)

## Checklist before requesting a review

- [x] You have performed a self-review of your own code
- [x] You are using approved terminology
- [x] Your changes have been tested locally
- [x] Your code builds clean without any errors or warnings

### CODE REVIEW

File: src/ui/style.css [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+.xar-social-icon {
+  margin: 0 auto;
+  width: 24px;
+  height: 24px;
+}
+
+.xar-container {
+  padding: 30px 30px;
+  min-width: 325px;
+  max-width: 325px;
```
Comment (shrinathprabhu) [inline src/ui/style.css]: Can be width if both are same

```css
width: 325px
```

File: src/ui/style.css [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+  --inputShadow: inset -2px -2px 4px rgb(57 57 57 / 44%),
+    inset 5px 5px 10px rgb(11 11 11 / 50%);
+}
+
+#xar-modal {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  position: fixed;
+  z-index: 1;
```
Comment (shrinathprabhu) [inline src/ui/style.css]: Should we add a bigger z-index?

Comment (makylfang) [reply src/ui/style.css]: @shrinathprabhu Suggestion?

File: src/ui/style.css (hunk 1 above)
Comment (makylfang) [reply src/ui/style.css]: Fixed.

File: src/ui/style.css (hunk 2 above)
Comment (shrinathprabhu) [reply src/ui/style.css]: Something big like 99999, so parent app's elements don't float on top of the modal

Comment (makylfang) [reply src/ui/style.css]: Updated.

**Review by shrinathprabhu — APPROVED** [review]

**Pull request merged**